underfs/s3/src/main/java/tachyon/underfs/s3/S3UnderFileSystem.java

```diff
@@ -0,0 +1,520 @@
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.underfs.s3;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jets3t.service.S3Service;
+import org.jets3t.service.ServiceException;
+import org.jets3t.service.impl.rest.httpclient.RestS3Service;
+import org.jets3t.service.model.S3Object;
+import org.jets3t.service.model.StorageObject;
+import org.jets3t.service.security.AWSCredentials;
+import org.jets3t.service.utils.Mimetypes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.underfs.UnderFileSystem;
+import tachyon.util.CommonUtils;
+
+/**
+ * Under file system implementation for S3 using the Jets3t library.
+ */
+public class S3UnderFileSystem extends UnderFileSystem {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private static final String FOLDER_SUFFIX = "_$folder$";
+  private static final String PATH_SEPARATOR = "/";
+
+  private final S3Service mClient;
+  private final String mBucketName;
+
+  public S3UnderFileSystem(String bucketName, TachyonConf tachyonConf) {
+    super(tachyonConf);
+    AWSCredentials awsCredentials =
+        new AWSCredentials(tachyonConf.get("fs.s3n.awsAccessKeyId", null), tachyonConf.get(
+            "fs.s3n.awsSecretAccessKey", null));
+    mBucketName = bucketName;
+    mClient = new RestS3Service(awsCredentials);
+  }
+
+  @Override
+  public void close() throws IOException {
+  }
+
+  @Override
+  public void connectFromMaster(TachyonConf conf, String hostname) {
+    // Authentication is taken care of in the constructor
+  }
+
+  @Override
+  public void connectFromWorker(TachyonConf conf, String hostname) {
+    // Authentication is taken care of in the constructor
+  }
+
+  @Override
+  public OutputStream create(String path) throws IOException {
+    if (mkdirs(getParentKey(path), true)) {
+      return new S3OutputStream(mBucketName, stripPrefix(path), mClient);
+    }
+    return null;
+  }
+
+  // Same as create(path)
+  @Override
+  public OutputStream create(String path, int blockSizeByte) throws IOException {
+    return create(path);
```

Comment:
Considering LOG.error (or LOG.warn?) here to indicate blockSizeByte is not used, before return create(path).